# Module 3 - Measures of Dispersion

* Measures of Dispersion describe how spread out or scattered the values in a dataset are around a central value such as the mean.

*  While Mean, Median, and Mode help us understand the center of the data, Measures of Dispersion help us understand how much the values vary.

### Why is it Important?

- Helps us understand data variability.
- Helps compare the consistency of different datasets.
- Helps identify unusual values or outliers.
- Supports data analysis in business and AI/ML.

### Topics Covered

- Range
- Variance
- Standard Deviation
- Quartiles
- Percentiles
- Interquartile Range (IQR)
- Coefficient of Variation

## 1. Range

### Definition

- Range is a measure of the overall spread of values in a dataset.
- It represents the difference between the highest and lowest values.

### Understand the Concept

- Range tells us how far the data extends from its smallest value to its largest value.
- A larger range means the values are more spread out, while a smaller range means they are closer together.
### Formula

$$
Range = Maximum\ Value - Minimum\ Value
$$

### Numerical Example

Consider the marks:

45, 50, 55, 60, 65, 70, 75, 80, 85, 90

- Maximum value = 90
- Minimum value = 45
- Range = 90 - 45 = 45

Therefore, the range is 45.

### Real-Life Example

A teacher wants to know the difference between the highest and lowest marks in a class.

If the highest mark is 90 and the lowest mark is 45:

Range = 90 - 45 = 45 marks.

### AI/ML Example

Range can help us understand how much a numerical feature varies.

For example, if customer ages range from 18 to 60, the range is 42 years.

### Interpretation

The range is 45, which means the difference between the highest and lowest marks is 45.

In [1]:
marks = [45, 50, 55, 60, 65, 70, 75, 80, 85, 90]

range_value = max(marks) - min(marks)

print("Range:", range_value)

Range: 45


## 2. Variance

### Definition

- Variance is a measure of how much the values in a dataset vary around the mean.
- It shows the amount of overall variation present in the data.

### Understand the Concept

- Variance looks at how far each value is from the mean and measures the overall spread.
- A smaller variance means the values are closer to the mean, while a larger variance means they are more spread out.
### Formula

$$
Population\ Variance = \frac{\sum(x-\mu)^2}{N}
$$

Where:

- x = each value
- μ = mean
- N = number of values

### Demonstrate the Concept

#### Numerical Example

Data: 2, 4, 6

Mean = (2 + 4 + 6) / 3 = 4

Differences from mean:

- 2 - 4 = -2
- 4 - 4 = 0
- 6 - 4 = 2

Squared differences:

4, 0, 4

Variance = (4 + 0 + 4) / 3

Variance = 2.67

#### Real-Life Example

A teacher compares the marks of two classes. If the marks in one class are close to the average, the variance is low. If the marks are very different from the average, the variance is high.

#### AI/ML Example

Variance helps us understand how much a feature changes in a dataset. It can also help identify features with very little variation.

### Interpretation

The variance is approximately 2.67, which shows the amount of spread of the values around the mean.

In [3]:
import numpy as np

marks = np.array([2, 4, 6])

print("Mean:", np.mean(marks))
print("Variance:", np.var(marks))

Mean: 4.0
Variance: 2.6666666666666665


## 3. Standard Deviation

### Definition

- Standard deviation is a measure of the typical spread of values around the mean.
- It shows how closely or widely the values are distributed around the average.

### Understand the Concept

- Standard deviation tells us approximately how far the values tend to be from the mean.
- A smaller standard deviation means the values are more consistent, while a larger standard deviation means they are more spread out.

### Formula

$$
Standard\ Deviation = \sqrt{Variance}
$$

### Demonstrate the Concept

#### Numerical Example

Data: 2, 4, 6

Mean = 4

Variance = 2.67

Standard Deviation = √2.67

Standard Deviation ≈ 1.63

#### Real-Life Example

A teacher can use standard deviation to understand whether students' marks are close to the average or widely spread.

#### AI/ML Example

Standard deviation helps us understand the spread of features. It is also useful when scaling data before training a Machine Learning model.

### Interpretation

The standard deviation is approximately 1.63. This means the values generally vary around the mean by about 1.63.

In [4]:
print("Mean:", np.mean(marks))
print("Variance:", np.var(marks))
print("Standard Deviation:", np.std(marks))

Mean: 4.0
Variance: 2.6666666666666665
Standard Deviation: 1.632993161855452


## 4. Quartiles

### Definition

- Quartiles are statistical measures used to describe how values are distributed within an ordered dataset.
- They divide the data into four sections to show the position and distribution of values.

### Understand the Concept

- Quartiles help us understand how the data is distributed from the lowest values to the highest values.
- Q1, Q2, and Q3 help identify the lower, middle, and upper parts of the dataset.

Q1 = 25th percentile

Q2 = 50th percentile or Median

Q3 = 75th percentile

### Formula

Q1 = 25th percentile

Q2 = 50th percentile

Q3 = 75th percentile

### Demonstrate the Concept

#### Numerical Example

Data:

10, 20, 30, 40, 50, 60, 70, 80

Q1 shows the value below which about 25% of the data falls.

Q2 is the median of the data.

Q3 shows the value below which about 75% of the data falls.

#### Real-Life Example

A company can use quartiles to divide employee salaries into four groups and understand how salaries are distributed.

#### AI/ML Example

In AI/ML, quartiles can help understand the distribution of a feature and identify unusual values.

### Interpretation

Q1 represents the lower part of the data, Q2 represents the middle, and Q3 represents the upper part of the data.

In [5]:
salaries = np.array([20000, 25000, 30000, 35000, 40000, 45000, 50000, 60000])

q1 = np.percentile(salaries, 25)
q2 = np.percentile(salaries, 50)
q3 = np.percentile(salaries, 75)

print("Q1:", q1)
print("Q2:", q2)
print("Q3:", q3)

Q1: 28750.0
Q2: 37500.0
Q3: 46250.0


## 5. Percentiles

## 5. Percentiles

### Definition

- Percentiles are statistical measures that describe the relative position of a value within a dataset.
- They indicate the value at or below which a specified percentage of observations falls.

### Understand the Concept

- Percentiles help us understand where a particular value stands compared with the rest of the data.
- For example, the 75th percentile means that about 75% of the observations are at or below that value.

### Demonstrate the Concept

#### Numerical Example

Data:

10, 20, 30, 40, 50, 60, 70, 80

The 25th percentile is 27.5.

This means about 25% of the data is at or below 27.5.

#### Real-Life Example

A student scores 85 marks in an exam. If the student is in the 90th percentile, it means the student performed better than about 90% of the students.

#### AI/ML Example

Percentiles can be used in AI/ML to understand the distribution of numerical features and identify unusually high or low values.

### Interpretation

The percentile tells us where a particular value stands compared with the rest of the dataset.

In [6]:
marks = np.array([10, 20, 30, 40, 50, 60, 70, 80])

p25 = np.percentile(marks, 25)
p50 = np.percentile(marks, 50)
p75 = np.percentile(marks, 75)

print("25th Percentile:", p25)
print("50th Percentile:", p50)
print("75th Percentile:", p75)

25th Percentile: 27.5
50th Percentile: 45.0
75th Percentile: 62.5


## 6. Interquartile Range (IQR)

### Definition

- Interquartile Range (IQR) is a measure of the spread of the middle portion of a dataset.
- It represents the difference between the third quartile (Q3) and the first quartile (Q1).

### Understand the Concept

- IQR focuses only on the middle 50% of the data.
- It is useful for understanding the central spread of the data and identifying possible outliers.
### Formula

$$
IQR = Q3 - Q1
$$

### Demonstrate the Concept

#### Numerical Example

Data:

10, 20, 30, 40, 50, 60, 70, 80

Q1 = 27.5

Q3 = 62.5

IQR = 62.5 - 27.5

IQR = 35

Therefore, the IQR is 35.

#### Real-Life Example

A company checks employee salaries. IQR helps the company understand the spread of the middle 50% of salaries.

Very high or very low salaries can also be checked as possible outliers.

#### AI/ML Example

IQR is commonly used to detect outliers in Machine Learning data. It can help find unusual values before training a model.

### Interpretation

An IQR of 35 means the middle 50% of the data is spread across a range of 35 units.

In [7]:
marks = np.array([10, 20, 30, 40, 50, 60, 70, 80])

q1 = np.percentile(marks, 25)
q3 = np.percentile(marks, 75)

iqr = q3 - q1

print("Q1:", q1)
print("Q3:", q3)
print("IQR:", iqr)

Q1: 27.5
Q3: 62.5
IQR: 35.0


In [8]:
lower_bound = q1 - (1.5 * iqr)
upper_bound = q3 + (1.5 * iqr)

outliers = [x for x in marks if x < lower_bound or x > upper_bound]

print("Lower Bound:", lower_bound)
print("Upper Bound:", upper_bound)
print("Outliers:", outliers)

Lower Bound: -25.0
Upper Bound: 115.0
Outliers: []


## 7. Coefficient of Variation

### Definition

- Coefficient of Variation (CV) is a measure of relative variability in a dataset.
- It compares the standard deviation with the mean and expresses the result as a percentage.

### Understand the Concept

- CV helps us compare the consistency of datasets even when their values or scales are different.
- A lower CV indicates less variation relative to the mean, while a higher CV indicates greater relative variation.


### Formula

$$
CV = \frac{Standard\ Deviation}{Mean} \times 100
$$

### Demonstrate the Concept

#### Numerical Example

Mean = 50

Standard Deviation = 5

CV = (5 / 50) × 100

CV = 10%

Therefore, the coefficient of variation is 10%.

#### Real-Life Example

Two machines produce the same type of product. We can use CV to compare how consistent their production is.

The machine with the lower CV has more consistent output.

#### AI/ML Example

In AI/ML, CV can help compare the relative variation of different numerical features, even when their scales are different.

### Interpretation

A lower CV indicates that the data has less variation relative to its mean.

In [10]:
import numpy as np

machine_a = [48, 50, 52, 49, 51]
machine_b = [40, 50, 60, 45, 55]

cv_a = (np.std(machine_a) / np.mean(machine_a)) * 100
cv_b = (np.std(machine_b) / np.mean(machine_b)) * 100

print("Machine A CV:", round(cv_a, 2), "%")
print("Machine B CV:", round(cv_b, 2), "%")

Machine A CV: 2.83 %
Machine B CV: 14.14 %


## Quick Comparison

| Measure | What it tells us | Common Use |
|---|---|---|
| Range | Difference between the highest and lowest values | Quick view of total spread |
| Variance | How much values differ from the mean | Measuring data variation |
| Standard Deviation | Typical spread around the mean | Measuring consistency |
| Quartiles | Divide data into four parts | Understanding data distribution |
| Percentiles | Show the position of values in a dataset | Comparing scores and values |
| IQR | Spread of the middle 50% | Detecting possible outliers |
| Coefficient of Variation | Variation compared with the mean | Comparing consistency between datasets |

## Conclusion

- Measures of Dispersion help us understand how much the values in a dataset vary.
- Range gives the difference between the highest and lowest values.
- Variance and Standard Deviation show how the data is spread around the mean.
- Quartiles and Percentiles help us understand the position and distribution of values.
- IQR helps measure the spread of the middle 50% and identify possible outliers.
- Coefficient of Variation helps compare the consistency of different datasets.
- These concepts are useful for understanding data in business and AI/ML.